pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -0,0 +1,205 @@
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/informers"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/record"
+	utiltesting "k8s.io/client-go/util/testing"
+	controllervolumetesting "k8s.io/kubernetes/pkg/controller/volume/attachdetach/testing"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/csi"
+	"k8s.io/kubernetes/pkg/volume/gcepd"
+	volumetest "k8s.io/kubernetes/pkg/volume/testing"
+	volumetesting "k8s.io/kubernetes/pkg/volume/testing"
+	volumetypes "k8s.io/kubernetes/pkg/volume/util/types"
+	"os"
+	"testing"
+	"time"
+
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+const gcePdVolumePluginName = "kubernetes.io/gce-pd"
+
+type testcase struct {
+	name                  string
+	isCsiMigrationEnabled bool
+	expectedPluginName    string
+}
+
+// this method just tests the volume plugin name that's used in CompleteFunc, the same plugin is also used inside the
+// generated func so there is no need to test the plugin name that's used inside generated function
+func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
+
+	testcases := []testcase{
+		{
+			name:                  "csi migration disabled",
+			isCsiMigrationEnabled: false,
+			expectedPluginName:    gcePdVolumePluginName,
+		},
+		{
+			name:                  "csi migration enabled",
+			isCsiMigrationEnabled: true,
+			expectedPluginName:    csi.CSIPluginName,
+		},
+	}
+
+	for _, tc := range testcases {
+		if tc.isCsiMigrationEnabled {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, true)()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationGCE, true)()
+		}
+		volumePluginMgr, plugin, tmpDir := initTestPlugins(t)
+		defer os.RemoveAll(tmpDir)
+
+		operationGenerator := getTestOperationGenerator(volumePluginMgr)
+
+		pdName := "pd-volume"
+		podName := "pod-1"
+		pod := getTestPodWithGCEPD(podName, pdName)
+		volumeToUnmount := getTestVolumeToUnmount(podName, pdName, pod)
+		expectedPluginName := tc.expectedPluginName
+
+		if tc.isCsiMigrationEnabled {
+			csiSpec, err := translateSpec(volumeToUnmount.VolumeSpec)
+			if err != nil {
+				t.Fatalf("Can't translate volume to CSI")
+			}
+			expectedPluginName = util.GetFullQualifiedPluginNameForVolume(tc.expectedPluginName, csiSpec)
+			_, mapper_error := plugin.(volume.BlockVolumePlugin).NewBlockVolumeMapper(csiSpec, pod, volume.VolumeOptions{})
+			if mapper_error != nil {
+				t.Fatalf("mapper error: %v\n", mapper_error)
+			}
+		}
+
+		unmapVolumeFunc, e := operationGenerator.GenerateUnmapVolumeFunc(volumeToUnmount, nil)
+		if e != nil {
+			t.Fatalf("Error occured while generating unmapVolumeFunc: %v", e)
+		}
+
+		var ee error
+		unmapVolumeFunc.CompleteFunc(&ee)
+
+		metric_family_name := "storage_operation_status_count"
```

Comment:
recommend to use go style for  variable declarations
[https://golang.org/doc/effective_go.html#mixed-caps](https://golang.org/doc/effective_go.html#mixed-caps)

Reply:
fixed, thanks!